src/NativeScript/Marshalling/Reference/ExtVectorTypeInstance.cpp

```diff
@@ -0,0 +1,121 @@
+//
+//  ExtVectorTypeInstance.cpp
+//  NativeScript
+//
+//  Created by Teodor Dermendzhiev on 30/01/2018.
+//
+
+#include "ExtVectorTypeInstance.h"
+#include "FFISimpleType.h"
+#include "IndexedRefInstance.h"
+#include "Interop.h"
+#include "PointerInstance.h"
+#include "RecordConstructor.h"
+#include "ReferenceInstance.h"
+#include "ReferenceTypeInstance.h"
+#include "ffi.h"
+
+namespace NativeScript {
+using namespace JSC;
+typedef ReferenceTypeInstance Base;
+
+const ClassInfo ExtVectorTypeInstance::s_info = { "ExtVectorTypeInstance", &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(ExtVectorTypeInstance) };
+
+JSValue ExtVectorTypeInstance::read(ExecState* execState, const void* buffer, JSCell* self) {
+    const void* data = buffer; //*reinterpret_cast<void* const*>(buffer);
+
+    if (!data) {
+        return jsNull();
+    }
+
+    GlobalObject* globalObject = jsCast<GlobalObject*>(execState->lexicalGlobalObject());
+    ExtVectorTypeInstance* referenceType = jsCast<ExtVectorTypeInstance*>(self);
+
+    PointerInstance* pointer = jsCast<PointerInstance*>(globalObject->interop()->pointerInstanceForPointer(execState->vm(), const_cast<void*>(data)));
+    return IndexedRefInstance::create(execState->vm(), globalObject, globalObject->interop()->extVectorInstanceStructure(), referenceType->innerType(), pointer);
+}
+
+void ExtVectorTypeInstance::write(ExecState* execState, const JSValue& value, void* buffer, JSCell* self) {
+    ExtVectorTypeInstance* referenceType = jsCast<ExtVectorTypeInstance*>(self);
+
+    if (value.isUndefinedOrNull()) {
+        // *reinterpret_cast<void**>(buffer) = nullptr;
+        buffer = nullptr;
+        return;
+    }
+
+    if (IndexedRefInstance* reference = jsDynamicCast<IndexedRefInstance*>(execState->vm(), value)) {
+        if (!reference->data()) {
+            GlobalObject* globalObject = jsCast<GlobalObject*>(execState->lexicalGlobalObject());
+            reference->createBackingStorage(execState->vm(), globalObject, execState, referenceType->innerType());
+        }
+    }
+
+    bool hasHandle;
+    JSC::VM& vm = execState->vm();
+    void* handle = tryHandleofValue(vm, value, &hasHandle);
+    if (!hasHandle) {
+        JSC::VM& vm = execState->vm();
+        auto scope = DECLARE_THROW_SCOPE(vm);
+
+        JSValue exception = createError(execState, WTF::ASCIILiteral("Value is not a reference."));
+        scope.throwException(execState, exception);
+        return;
+    }
+
+    //*reinterpret_cast<void**>(buffer) = handle;
+    buffer = handle;
+}
+
+const char* ExtVectorTypeInstance::encode(VM& vm, JSCell* cell) {
+    ExtVectorTypeInstance* self = jsCast<ExtVectorTypeInstance*>(cell);
+
+    if (!self->_compilerEncoding.empty()) {
+        return self->_compilerEncoding.c_str();
+    }
+
+    self->_compilerEncoding = "^";
```

Comment:
According to the [Type Encodings documentation article](https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/ObjCRuntimeGuide/Articles/ocrtTypeEncodings.html)
>The type code for an array is enclosed within square brackets; the number of elements in the array is specified immediately after the open bracket, before the array type. For example, an array of 12 pointers to floats would be encoded as:
[12^f]

I think that `IndexedRefTypeInstance`'s and `ExtVectorTypeInstance`'s compiler encodings should both be changed accordingly.